src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient.cs

```diff
@@ -34,128 +35,193 @@ public ImageStoreClient(ServiceFabricHttpClient httpClient)
             long? serverTimeout = 60,
             CancellationToken cancellationToken = default(CancellationToken))
         {
             contentPath.ThrowIfNull(nameof(contentPath));
-            serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
-            var requestId = Guid.NewGuid().ToString();
-            var url = "ImageStore/{contentPath}";
-            url = url.Replace("{contentPath}", Uri.EscapeDataString(contentPath));
-            var queryParams = new List<string>();
-            
-            // Append to queryParams if not null.
-            serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
-            queryParams.Add("api-version=6.2");
-            url += "?" + string.Join("&", queryParams);
-            
-            HttpRequestMessage RequestFunc()
+
+            await LoadImageStoreConnectionString();
+            if (isLocalStore)
             {
-                var request = new HttpRequestMessage()
-                {
-                    Method = HttpMethod.Get,
-                };
-                return request;
+                var path = Path.Combine(imageStorePath, contentPath);
+                var directories = Directory.EnumerateDirectories(path, "*", SearchOption.AllDirectories).Select(d => new FolderInfo(d, (long)Directory.EnumerateFiles(d).Count())).ToList();
+                var files = Directory.EnumerateFiles(path, "*", SearchOption.AllDirectories).Select(f => new System.IO.FileInfo(f)).Select(f => new Common.FileInfo(f.Length.ToString(), new FileVersion(), f.LastWriteTimeUtc, f.FullName.Replace(imageStorePath, ""))).ToList();
+                return new ImageStoreContent(files, directories);
             }
+            else
+            {
+                serverTimeout?.ThrowIfOutOfInclusiveRange("serverTimeout", 1, 4294967295);
+                var requestId = Guid.NewGuid().ToString();
+                var url = "ImageStore/{contentPath}";
+                url = url.Replace("{contentPath}", Uri.EscapeDataString(contentPath));
+                var queryParams = new List<string>();
+
+                // Append to queryParams if not null.
+                serverTimeout?.AddToQueryParameters(queryParams, $"timeout={serverTimeout}");
+                queryParams.Add("api-version=6.2");
+                url += "?" + string.Join("&", queryParams);
+
+                HttpRequestMessage RequestFunc()
+                {
+                    var request = new HttpRequestMessage()
+                    {
+                        Method = HttpMethod.Get,
+                    };
+                    return request;
+                }
 
-            return this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
+                return await this.httpClient.SendAsyncGetResponse(RequestFunc, url, ImageStoreContentConverter.Deserialize, requestId, cancellationToken);
```

Comment:
>return [](start = 16, length = 6)

This can just be return, 

[https://stackoverflow.com/questions/21033150/any-difference-between-await-task-run-return-and-return-task-run](https://stackoverflow.com/questions/21033150/any-difference-between-await-task-run-return-and-return-task-run)

Reply:
It can not, since it's an async method and it has to be async since it retrieves the connection string.